kinode/packages/kns_indexer/kns_indexer/src/lib.rs

```diff
@@ -108,29 +290,31 @@ fn main(our: Address, mut state: State) -> anyhow::Result<()> {
 
     // sub_id: 2
     let notes_filter = eth::Filter::new()
-        .address(state.contract_address)
+        .address(state.get_contract_address())
```

Comment:
optimization: get contract address once at beginning and use it multiple times? calling kv.get() over and over is a lot, no?